src/integrationTest/cloudformation/templateRegistryManager.test.ts

```diff
@@ -48,6 +50,12 @@ describe('CloudFormation Template Registry Manager', async () => {
         dir++
     })
 
+    after(() => {
+        if (removeLoggerAfterTest) {
+            setLogger(undefined)
+        }
```

Comment:
with the boolean, `TestLogger` remains as the global logger which will be used by other tests?

Reply:
It's the opposite. If this test suite added the logger, it removes it, so that other test suites do not have this pre-existing logger as a side effect.

Reply:
would be clear to check that the type is `TestLogger` rather than maintaining a boolean

Reply:
I've revised it to check for the instantiated object

Reply:
oh, hmm. I was hoping to avoid the extra state. Wouldn't it work to check the type of `getLogger()` ? Something like `typeof getLogger() == TestLogger`.

 Or that doesn't work because TS is structurally typed? Then `Logger` would need a discriminator field, which still seems preferable to the ad-hoc discriminator field we've added in this test module.

Reply:
> Then Logger would need a discriminator field, which still seems preferable to the ad-hoc discriminator field we've added in this test module.

Defensive programming in the event some other test was careless in its logger management. This test suite only cares to clean up its own logger, if it assigned one.

> Side-note: it's still not clear to me why we need the if-check at all: what is the harm in always doing setLogger(undefined) here? The before() block is guaranteed to run...

The integration tests are intended to run against a running instance of the Toolkit. The integration tests share the experimental VS Code session, and don't get a freshly activated toolkit for each integration test suite. (To do that, we'd have to launch different test suites from the command line). Unconditionally clearing the logger could inadvertently clear the extensions real logger.

Reply:
> Unconditionally clearing the logger could inadvertently clear the extensions real logger.

Why doesn't this test suite unconditionally _set_ the logger, instead of the try-catch dance? Then we can unconditionally un-set it.

Reply:
Each of the test suites (top level `describe` statements) that are run from within `src/integrationTest` are intended to be run against VS Code, most of them with the extension activated (for example, sam.test.ts makes sure the extension is activated prior to running tests). Unconditionally setting the logger would corrupt the extensions logger, if the extension has already been activated.

This file seems to be a bit of an outlier, in that it exercises parts of the code based on VS Code's workspace, and doesn't care if the extension is activated. It is also the first suite to run if I'm not mistaken, but we should make no assumptions that some future test suite will be run before this one.